File: radar-commons/src/main/java/org/radarbase/util/TimedValue.java
package org.radarbase.util;

import java.util.Objects;

public class TimedValue<T> implements TimedVariable {
    public final T value;
    private final long expiry;

    public TimedValue(T value, long validity) {
        expiry = System.currentTimeMillis() + validity * 1000L;
        this.value = Objects.requireNonNull(value);
    }

    @Override
    public boolean isExpired() {
        return expiry < System.currentTimeMillis();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        TimedValue<?> other = (TimedValue<?>)o;
        return value.equals(other.value)
                && expiry == other.expiry;
    }

    @Override
    public int hashCode() {
        return value.hashCode();
    }
}

File: radar-commons/src/main/java/org/radarbase/util/TimedValue.kt
package org.radarbase.util

class TimedValue<T: Any>(
    val value: T,
    private val cacheConfig: CacheConfig,
) : TimedVariable(cacheConfig) {
    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        other as TimedValue<*>
        return value == other.value && expiry == other.expiry
    }

    override fun hashCode(): Int = value.hashCode()
}
